home/.chezmoiscripts/universal/run_onchange_after_57-elastic-agent.sh.tmpl

```diff
@@ -0,0 +1,29 @@
+{{- if and (ne .host.distro.family "windows") (stat (joinPath .host.home ".config" "age" "chezmoi.txt")) (or (stat (joinPath .chezmoi.sourceDir ".chezmoitemplates" "secrets" "ELASTIC_TOKEN")) (env "ELASTIC_TOKEN")) -}}
+#!/usr/bin/env bash
+# @file Netdata
+# @brief Enrolls Elastic Agent with Elastic Fleet Server
+# @description
+#     This script enrolls Elastic Agent with Elastic Fleet Server if Elastic Agent is installed, the `ELASTIC_TOKEN` is provided, and the
+#     `elastic_fleet_server` is defined.
+#
+#     Fleet provides a web-based UI in Kibana for centrally managing Elastic Agents and their policies. Fleet Server is the mechanism
+#     to connect Elastic Agents to Fleet. All communication between the Fleet UI and Fleet Server happens through Elasticsearch.
+#
+#     [Integrations](https://www.elastic.co/guide/en/fleet/current/integrations.html) and [Policies](https://www.elastic.co/guide/en/fleet/current/agent-policy.html)
+#     are configured using Fleet UI in Kibana. The [enrollment-token](https://www.elastic.co/guide/en/fleet/current/install-fleet-managed-elastic-agent.html)
```

Comment:
Hey @enggnr -- looks like Fleet needs to be installed manually outside of the `software.yml` for now. See: https://www.elastic.co/guide/en/fleet/7.17/install-fleet-managed-elastic-agent.html -- use the Linux installer for all Linux systems and be sure to enable it as a service.

Reply:
Yes, at least one Fleet Server needs to be available before running this script. Fleet server is enabled using the elastic agent installer, and there can be more than 1 Fleet Server. This is what the script is doing.

---
When the script enrolls the agent, it expects a Fleet Server URL and an Enrollment token to be provided. The installation process sets up the service automatically.

The URL that is provided as an input to the script points to agent where the Fleet server is running, which would be the value of `elasticFleetServer`.

The Token is obtained using the UI, which is possible only after a Fleet Server is available. It couldn't find a CLI method to do this.

While it is possible to enroll agents without a Fleet server, such agents are not visible in Fleet and are not managed using Fleet. The documentation for these are a bit lacking - focus appears to be on Fleet. The steps to obtain the token for such agents is different, command syntax is different and it was hard to find the info about these agents in Elastic UI. This was why I restricted the script to work only with Fleet based agents. It could be that support for these agents may be removed in a future release.